pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -279,13 +282,14 @@ func (c *Controller) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(cKey)
 
-	err := c.syncService(cKey.(string))
-	c.handleErr(err, cKey)
+	logger := klog.FromContext(context.TODO())
```

Comment:
since `processNextWorkItem()` is called inside `worker()` (so worker should also take a logger.) but `c.worker` is passed into `wait.Until()` [here](https://github.com/kubernetes/kubernetes/pull/115295/files/cb0dc5e94d922213e3d2f3d74a7176a98b93c0a0#diff-8765679aa8ca8cc500a6dc038b8707cbc29ddc2d2f94f1da911e2cd5e4df5816R263)

I believe it will require more changes in vendor/staging, wdyt?

Reply:
Thanks. Addressed 💪